src/Parser/HolyPriest/Modules/PriestCore/MasteryBreakdown.js

```diff
@@ -0,0 +1,200 @@
+import React from 'react';
+import { STATISTIC_ORDER } from 'Main/StatisticBox';
+import ExpandableStatisticBox from 'Main/ExpandableStatisticBox';
+import SpellIcon from 'common/SpellIcon';
+import { formatPercentage, formatNumber } from 'common/format';
+
+// dependencies
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import SPELLS from 'common/SPELLS';
+import Module from 'Parser/Core/Module';
+import { ABILITIES_THAT_TRIGGER_MASTERY } from '../../Constants';
+
+class MasteryBreakdown extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  }
+
+  _tickMode = {};
+  _healVal = {};
+  _maxHealVal = {};
+  _masteryActive = {};
+  effectiveHealDist = {};
+  effectiveOverhealDist = {};
+  _eHDbyPlayer = {}; // for potential future features
+
+  healing = 0;
+
+  effectiveHealDistPerc = [];
+
+  on_finished() {
+    // There's likely a far better way to do this, but my 2AM brain couldn't find it
+    let total = 0;
+
+    let spell;
+    for (spell in this.effectiveHealDist) {
+      total += this.effectiveHealDist[spell];
+    }
+
+    const eHDPerc = {};
+    for (spell in this.effectiveHealDist) {
+      eHDPerc[spell] = this.effectiveHealDist[spell] / total;
+    }
+
+    const eOHD = {};
+    for (spell in this.effectiveHealDist) {
+      eOHD[spell] = this.effectiveOverhealDist[spell] / (this.effectiveHealDist[spell] + this.effectiveOverhealDist[spell]);
+    }
+
+    // Since JS objects lack order, we need to convert our dictionary to an array
```

Comment:
If it's not necessary for the data handling it's customary to wait with ordering data until the rendering so that the renderer can decide how the data should be sorted (and perhaps allow users to alter how a table is sorted).